privacyidea/api/auth.py

```diff
@@ -289,6 +288,7 @@ def get_auth_token():
         log.info("Local admin '{0!s}' successfully logged in.".format(username))
         # This admin is not in the default realm!
         realm = ""
+        user_obj = User()
```

Comment:
To my understanding you are cleaning up the user_obj.
This could be if there is a user in the default realm with the same login name as a local admin.

However, I *think* I remeber we had problems in the past in this case, why we did a "late" user resolving (which would contradict the before_request).

I remember we had problems in this case, the LDAP is not accessable and due to an early user_obj creating the login for the local admin failed or timed out.
So if you move the user_obj creation to the beginning of this function, the code might look cleaner, but we need to verify that we do not reintroduce this problem again! This is why I am so hesitant about this.

Reply:
The creation of the `User` object in the `before_request` was added for pre-event handler.
This failed for an admin log-in if the resolver was broken. Since #2375 the log-in doesn't fail but if a resolver in the default realm is broken, we still need to wait for the timeout.
We could check in the `before_request` if the user is an internal admin (e.g. a `username` without a realm) and then bypass the creation of the `User` object.
I still need to check what happens if there is a user with the same name in the default realm.

Reply:
I've checked that the internal admin check still works despite a user with a same name in the default realm. Also the admin can still log-in if the resolver doesn't work, it just takes time to wait for the timeout from the resolver.

Reply:
Also the local admin vs. user check works as expected.